macros/commands/generate_json_schema_test.sql

```diff
@@ -0,0 +1,101 @@
+{% macro generate_json_schema_test(node_name, column_name) %}
+    {% if target.type not in ['snowflake', 'bigquery'] %}
+      {% do print("JSON schema test generation is not supported for target: {}".format(target.type)) %}
+      {% do return(none) %}
+    {% endif %}
+
+    {% set node = elementary.get_node_by_name(node_name) %}
+    {% if node.resource_type not in ["source", "model"] %}
+      {% do print("Only sources and models are supported for this macro, supplied node type: '{}'".format(node.resource_type)) %}
+      {% do return(none) %}
+    {% endif %}
+
+    {% set node_relation = get_relation_from_node(node) %}
+    {% if not elementary.column_exists_in_relation(node_relation, column_name) %}
+      {% do print("Column '{}' does not exist in {} '{}'!".format(column_name, node.resource_type, node_name)) %}
+      {% do return(none) %}
+    {% endif %}
+
+    {% set elementary_database_name, elementary_schema_name = elementary.get_package_database_and_schema() %}
+
+    {% do node.config.update({"packages": ["genson"]}) %}
+    {% do node.update({'database': elementary_database_name, 'schema': elementary_schema_name}) %}
+    {% if node.resource_type == 'source' %}
+      {# Source nodes don't have alias, and submit_python_job expects it #}
+      {% do node.update({'alias': "jsonschemagen_{}_{}".format(node.source_name, node.name)}) %}
+    {% endif %}
+
+    {% set output_table = api.Relation.create(database=elementary_database_name, schema=elementary_schema_name,
+        identifier='json_schema_tmp__' ~ node.alias).quote(false, false, false) %}
+
+    {% set gen_json_schema_func = elementary.generate_json_schema_py_func(column_name) %}
+    {% set node_relation = node_relation.quote(false, false, false) %}
+    {% set compiled_py_code = adapter.dispatch('compile_py_code', 'elementary')(node_relation, gen_json_schema_func,
+                                                                                output_table, code_type='function') %}
+
+    {% do elementary.run_python(node, compiled_py_code) %}
+    {% set json_schema = elementary.result_value('select result from {}'.format(output_table)) %}
+    {% if json_schema == 'genson_not_installed' %}
+      {% do print("ERROR - the 'genson' python library is missing from your warehouse.") %}
```

Comment:
Maybe use `exceptions.raise_compiler_error()` here?
Probably thinking too long into the future, but we'd probably want the macro to return non-zero exit code so that it fails if it's in CI or something.

Reply:
done